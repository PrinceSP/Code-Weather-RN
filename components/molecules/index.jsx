import Header from './Header'
import CustomUnits from './CustomUnits'
import Checkbox from './CheckBox'
import HourlyWeather from './HourlyWeather'
import LocationPlace from './LocationPlace'
import SearchHeader from './SearchHeader'

export {Header,CustomUnits,Checkbox,HourlyWeather,LocationPlace,SearchHeader}
