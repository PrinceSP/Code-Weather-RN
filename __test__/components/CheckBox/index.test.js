import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import {Checkbox} from '../../../components';

describe('CheckBox Component',()=>{
  const updateFunction = jest.fn();
  const mockLabel = "Passed"
  const mockBoolean = true
  const eventData = {
    personal_feeling:false,
    own_station:false,
    local_weather:false,
    global_weather:false,
    other:false
  }

  it('renders correctly when checked',()=>{
    const component = render(
      <Checkbox label={mockLabel} onChange={updateFunction} checked={mockBoolean}/>
    )
    expect(screen.getByText(mockLabel)).toBeTruthy();
    expect(component).toMatchSnapshot()
  })

  it('renders correctly when unchecked',()=>{
    const component = render(
      <Checkbox label={mockLabel} onChange={()=>{}} checked={false}/>
    )
    expect(screen.getByText(mockLabel)).toBeTruthy();
    expect(component).toMatchSnapshot()
  })

  it('invokes onChange when pressed',()=>{
    const component = render(
      <Checkbox label={mockLabel} onChange={updateFunction} checked={mockBoolean}/>
    )
    fireEvent.press(screen.getByText(mockLabel),eventData)
    expect(updateFunction).toHaveBeenCalledWith(eventData)
  })

})
