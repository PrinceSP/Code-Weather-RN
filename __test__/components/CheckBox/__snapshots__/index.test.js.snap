// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
      },
      {
        "marginBottom": 20,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#a3a3a3",
          "borderRadius": 4,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 22,
        },
        {
          "backgroundColor": "#7866DC",
          "borderColor": "#7866DC",
        },
      ]
    }
  >
    <Text />
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    Passed
  </Text>
</View>
`;

exports[`CheckBox Component renders correctly when checked 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
      },
      {
        "marginBottom": 20,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#a3a3a3",
          "borderRadius": 4,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 22,
        },
        {
          "backgroundColor": "#7866DC",
          "borderColor": "#7866DC",
        },
      ]
    }
  >
    <Text />
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    Passed
  </Text>
</View>
`;

exports[`CheckBox Component renders correctly when unchecked 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
      },
      {
        "marginBottom": 20,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#a3a3a3",
          "borderRadius": 4,
          "borderWidth": 2,
          "height": 22,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 22,
        },
        false,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    Passed
  </Text>
</View>
`;
